src/cpp/handler/handlerengine.cpp

```diff
@@ -232,8 +232,8 @@ class InspectWorker : public Deferred
 			if(getSession && stateClient)
 			{
 				// determine session info
-				Deferred *d = SessionRequest::detectRulesGet(stateClient, requestData.uri.host().toUtf8(), requestData.uri.path(QUrl::FullyEncoded).toUtf8(), this);
-				finishedConnection[d] = d->finished.connect(boost::bind(&InspectWorker::sessionDetectRulesGet_finished, this, boost::placeholders::_1));
+				auto d = std::unique_ptr<Deferred>(SessionRequest::detectRulesGet(stateClient, requestData.uri.host().toUtf8(), requestData.uri.path(QUrl::FullyEncoded).toUtf8()));
```

Comment:
Needs to be a member to stay alive.

Reply:
I think a good way to keep this alive as long as needed is to use deferreds as such 
```
deferreds[d.get()] = std::move(d);
```

Reply:
maybe we can use a data structure similar to `deferreds` here
Also looking through the code, connections in InspectWorker and AcceptWorker are not erased from the map. maybe at the end of `doFinish()`

Reply:
yeah I think those classes simply clean up when they eventually destruct themselves, compared to the engine which is long lived and needs to regularly clean up as it goes